hudi-client/src/test/java/org/apache/hudi/common/HoodieTestDataGenerator.java

```diff
@@ -435,11 +439,46 @@ public HoodieRecord generateUpdateRecord(HoodieKey key, String commitTime) throw
         index = (index + 1) % numExistingKeys;
         kp = existingKeys.get(index);
       }
+      existingKeys.remove(kp);
```

Comment:
Shouldn't the remove be with the key rather than the value?

Reply:
yes, you are right. I figured this recently. working on the fix.